atcoder-problems-frontend/src/pages/Internal/VirtualContest/ShowContest/FirstAcceptedRow.tsx

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { ProblemId, UserId } from "../../../../interfaces/Status";
+import { ReducedProblemResult } from "./ResultCalcUtil";
+import { FirstAccepted, FirstAcceptedCell } from "./FirstAcceptedCell";
+
+interface FirstAcceptedRowProps {
+  start: number;
+  userIds: UserId[];
+  problemIds: string[];
+  resultsByUser: Map<UserId, Map<ProblemId, ReducedProblemResult>>;
+  showRating: boolean;
+}
+
+export const FirstAcceptedRow: React.FC<FirstAcceptedRowProps> = (props) => {
+  const { start, userIds, problemIds, resultsByUser, showRating } = props;
+
+  const fastestByProblem = new Map<string, FirstAccepted>();
+
+  resultsByUser.forEach((resultsByProblemId, userId) => {
+    if (!userIds.includes(userId)) {
+      return;
+    }
+
+    resultsByProblemId.forEach((result, problemId) => {
+      const currentFastest = fastestByProblem.get(problemId);
+      const time = result.lastUpdatedEpochSecond - start;
+
+      if (
+        (currentFastest === undefined && result.accepted) ||
+        (currentFastest && currentFastest.time > time && result.accepted)
+      ) {
```

Comment:
if文の中の両方の条件で `result.accepted` が必須なので、その条件を先に見てしまうようにするとif文の中がすっきりします。

```suggestion
      if (!result.accepted) {
          return;
      }

      const currentFastest = fastestByProblem.get(problemId);
      const time = result.lastUpdatedEpochSecond - start;
      if (!currentFastest || currentFastest.time > time) {
```